Sources/NIO/EventLoopFuture.swift

```diff
@@ -1325,3 +1325,81 @@ extension EventLoopFuture {
         return self
     }
 }
+
+// MARK: unwrap
+
+extension EventLoopFuture {
+    /// Unwrap an `EventLoopFuture` where its type parameter is an `Optional`.
+    ///
+    /// Unwrap a future returning a new `EventLoopFuture`. When the resolved future's value is `Optional.some(...)`
+    /// the new future is created with the identical value. Otherwise the `Error` passed in the `or` parameter
+    /// is thrown. For example:
+    /// ```
+    /// do {
+    ///     try promise.futureResult.unwrap(or: ErrorToThrow).wait()
+    /// } catch ErrorToThrow {
+    ///     ...
+    /// }
+    /// ```
+    ///
+    /// - parameters:
+    ///     - orError: the `Error` that is thrown when then resolved future's value is `Optional.none`.
+    /// - returns: an new `EventLoopFuture` with new type parameter `T` and the same value as the resolved
+    ///     future.
+    /// - throws: the `Error` passed in the `orError` parameter when the resolved future's value is `Optional.none`.
+    @inlinable
+    func unwrap<NewValue>(orError error: Error) -> EventLoopFuture<NewValue> where Value == Optional<NewValue> {
+        return self.flatMapThrowing { (value) throws -> NewValue in
+            guard let value = value else {
+                throw error
+            }
+            return value
+        }
+    }
+
+    /// Unwrap an `EventLoopFuture` where its type parameter is an `Optional`.
+    ///
+    /// Unwraps a future returning a new `EventLoopFuture` with either: the value passed in the `or`
+    /// parameter when the future resolved with value Optional.none, or the same value otherwise. For example:
+    /// ```
+    /// promise.futureResult.unwrap(or: 42).wait()
+    /// ```
+    ///
+    /// - parameters:
+    ///     - orReplace: the value of the returned `EventLoopFuture` when then resolved future's value is `Optional.some()`.
+    /// - returns: an new `EventLoopFuture` with new type parameter `T` and the value passed in the `orReplace` parameter.
+    @inlinable
+    func unwrap<NewValue>(orReplace replacement: NewValue) -> EventLoopFuture<NewValue> where Value == Optional<NewValue> {
+        return self.map { (value) -> NewValue in
+            guard let value = value else {
+                return replacement
+            }
+            return value 
+        }
+    }
+
+    /// Unwrap an `EventLoopFuture` where its type parameter is an `Optional`.
+    ///
+    /// Unwraps a future returning a new `EventLoopFuture` with either: the value returned by the closure passed in
+    /// the `or` parameter when the future resolved with value Optional.none, or the same value otherwise. For example:
+    /// ```
+    /// var x = 2
+    /// promise.futureResult.unwrap(or: { x * 2 }).wait()
```

Comment:
```suggestion
    /// promise.futureResult.unwrap(orElse: { x * 2 }).wait()
```

Reply:
Annoyingly I made all these changes and they got lost due to multitasking confusing :(